system/CodeIgniter.php

```diff
@@ -890,11 +895,6 @@ protected function startController()
 		{
 			throw PageNotFoundException::forControllerNotFound($this->controller, $this->method);
 		}
-		if (! method_exists($this->controller, '_remap') &&
-				! is_callable([$this->controller, $this->method], false))
```

Comment:
PHP 8 changes the behavior of `is_callable()` such that we need to pass it an instantiated object. Since we are about to get the controller instance anyways I moved this check after.

> The ability to call non-static methods statically has been removed. Thus is_callable() will fail when checking for a non-static method with a classname (must check with an object instance).

https://www.php.net/manual/en/migration80.incompatible.php

Reply:
Related to testing, I think we should change how we invoked phpunit's `assert*` methods. It is by design a static method and phpunit allows calling it non statically. However when I used phpstan's extra strict rules, I get 'Dynamic calls to static method assert** is not allowed'.

Reply:
We're doing it "the PHPUnit way": https://phpunit.readthedocs.io/en/9.3/assertions.html

I didn't even realize they were static. Is there some localized forwarder or something?

Reply:
I've looked into the code of TestCase and Assert and didn't find some wrapper to dynamic calls. Maybe phpunit is taking advantage of the way static methods can be called dynamically in php lt 8.